packages/lib/src/spatial-navigation/components/virtualizedList/helpers/getNumberOfItemsVisibleOnScreen.ts

```diff
@@ -0,0 +1,43 @@
+const getMinSizeOfItems = <T>({
+  data,
+  itemSize,
+}: {
+  data: T[];
+  itemSize: number | ((item: T) => number);
+}) => {
+  if (typeof itemSize === 'number') {
+    return itemSize;
+  }
+
+  let minSize: number | undefined = undefined;
+
+  for (let i = 0; i < data.length; i++) {
+    const item = data[i];
+    if (item !== undefined) {
+      const currentSize = itemSize(item);
+      if (minSize === undefined || currentSize < minSize) {
+        minSize = currentSize;
+      }
+    }
+  }
+
+  return minSize;
+};
+
+export const getNumberOfItemsVisibleOnScreen = <T>({
+  data,
+  listSizeInPx,
+  itemSize,
+}: {
+  data: T[];
+  listSizeInPx: number;
+  itemSize: number | ((item: T) => number);
+}) => {
+  const itemSizeToComputeRanges = getMinSizeOfItems({ data, itemSize });
```

Comment:
I feel like it's strange to use the max item size to compute this. What if we have 1 big item in the list, doesn't this mess up the virtualisation?

Is it well-documented that this like that might happen  and that we should use items of the same size preferably?

(or maybe I'm not aware about some improvements that you might have done )

Reply:
As I said in the other comment, the target is to use the size of the smallest item in the data, to anticipate worst case scenario. If we scroll through big items it won't be a problem because virtualization would have been tailored to handle little items 😄 
But as you said, my naming may induce confusion !

Reply:
(which is something to correct obviously)